tuplex/python/src/PythonDataSet.cc

```diff
@@ -105,6 +105,72 @@ namespace tuplex {
         }
     }
 
+    PythonDataSet PythonDataSet::sort(boost::python::list order, boost::python::list enumInOrder) {
+        assert(this->_dataset);
+        // error data set
+        if (_dataset->isError()) {
+            PythonDataSet pds;
+            pds.wrap(this->_dataset);
+            return pds;
+        }
+
+        // empty dataset
+        if (_dataset->isEmpty()) {
+            PythonDataSet pds;
+            pds.wrap(this->_dataset);
+            return pds;
+        }
+
+        // there are two possible modes how list can be given:
+        // 1.) selection using positive integers
+        // 2.) via strings
+        // ==> mixture also ok...
+        auto rowType = _dataset->schema().getRowType();
+        assert(rowType.isTupleType());
+        auto num_cols = rowType.parameters().size();
+
+        // go through list
+        std::vector<std::size_t> convOrder;
+        std::vector<std::size_t> convOrderEnum;
+        auto columns = _dataset->columns();
+
+        PyObject * listObj1 = order.ptr();
+        PyObject * listObj2 = enumInOrder.ptr();
+        for (unsigned i = 0; i < boost::python::len(order); ++i) {
+            PyObject * obj1 = PyList_GetItem(listObj1, i);
+            PyObject * obj2 = PyList_GetItem(listObj2, i);
+
+            // check type:
+            if (PyObject_IsInstance(obj1, reinterpret_cast<PyObject *>(&PyLong_Type)) &&
+                PyObject_IsInstance(obj2, reinterpret_cast<PyObject *>(&PyLong_Type))) {
+                int index = boost::python::extract<int>(order[i]);
+                int enumSort = boost::python::extract<int>(enumInOrder[i]);
+
+                if (index < 0 || index >= num_cols || enumSort <= 0 || enumSort > 6) {
+                    throw std::runtime_error("invalid index " + std::to_string(index)
```

Comment:
does this work? I.e., can we have a test which violates that constraint? (in py.test)